Example/src/MeasureExample.js

```diff
@@ -112,22 +112,51 @@ function Section({ title, children, height, contentHeight, z, show }) {
   );
 }
 
+function asyncMeasure(animatedRef) {
+  return new Promise((resolve, reject) => {
+    if (animatedRef && animatedRef.current) {
+      animatedRef.current.measure((x, y, width, height, pageX, pageY) => {
+        resolve({ x, y, width, height, pageX, pageY });
+      });
+    } else {
+      reject(new Error('measure: animated ref not ready'));
+    }
+  });
+}
+
 function SectionHeader({ title, animatedRef, contentHeight, show }) {
-  const handler = useAnimatedGestureHandler({
-    onActive: (_, ctx) => {
-      const height = measure(animatedRef).height;
-      if (contentHeight.value === 0) {
-        contentHeight.value = withTiming(height, {
-          duration: 500,
-          easing: Easing.bezier(0.25, 0.1, 0.25, 1),
-        });
-      } else {
-        contentHeight.value = withTiming(0, {
-          duration: 300,
-          easing: Easing.bezier(0.25, 0.1, 0.25, 1),
-        });
+  const applyMeasure = ({ x, y, width, height, pageX, pageY }) => {
+    if (contentHeight.value === 0) {
+      contentHeight.value = withTiming(height, {
+        duration: 500,
+        easing: Easing.bezier(0.25, 0.1, 0.25, 1),
+      });
+    } else {
+      contentHeight.value = withTiming(0, {
+        duration: 300,
+        easing: Easing.bezier(0.25, 0.1, 0.25, 1),
+      });
+    }
+  };
+
+  let onActiveImpl;
+  if (Platform.OS === 'web') {
+    onActiveImpl = async (_, ctx) => {
+      try {
+        applyMeasure(await asyncMeasure(animatedRef));
+      } catch (e) {
+        console.log(e);
```

Comment:
What happens if measure fails on native? Shouldn't we just let it fails instead of logging the exception?

Reply:
ok, we can rethrow c7ce9ee